opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/FacadeService.java

```diff
@@ -22,26 +23,37 @@
     private final ProtocolSelector selector;
     private final ServiceContextProvider provider;
     private final ProtocolResultHandler handler;
+    private final TransactionTemplate txTemplate;
 
     public CompletableFuture<FacadeResult<O>> execute(I request) {
-        ServiceContext<I> ctx = contextFor(request);
-        A protocol = selectAndSetProtocolTo(ctx);
-        CompletableFuture<Result<O>> result = execute(protocol, ctx);
+        ProtocolWithCtx<A, I> protocolWithCtx = txTemplate.execute(callback -> {
+            InternalContext<I> internalContext = contextFor(request);
+            A protocol = selectAndSetProtocolTo(internalContext);
+            ServiceContext<I> serviceContext = addRequestScopedFor(request, internalContext);
+            return new ProtocolWithCtx<>(protocol, serviceContext);
+        });
+        assert protocolWithCtx != null;
```

Comment:
Throw NullPointer exception instead of assertion - assertion has optional execution 
https://docs.oracle.com/javase/8/docs/technotes/guides/language/assert.html
`Enabling and Disabling Assertions`